### ISSUE TITLE
`pulumi new` not working for a template hosted in subgroups / projects on `gitlab.com`

### ISSUE THREAD

**Issue by yodamad** [pulumi/pulumi#17582]
### What happened?

I tried to create a new Pulumi project from a "private" template have made on `gitlab.com` (but which is available in a Public repository).

If I execute the command like explained in the `help` section, it doesn't work.

I cannot figure out why it doesn't work like explained in the documentation.

### Example

First, with command `pulumi new https://gitlab.com/yodamad-workshops/2024/devfest-toulouse/pulumi-template/`, I got the error `Authentication required`, even if the repository is public

Second, with command containing authentication credentials, `pulumi new https://<my_username>:<my_token>@gitlab.com/yodamad-workshops/2024/devfest-toulouse/pulumi-template/`, I got the error `repository not found`

### Output of `pulumi about`

CLI
Version      3.135.1
Go Version   go1.23.2
Go Compiler  gc

Host
OS       darwin
Version  14.7
Arch     x86_64

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
Hi @yodamad, thanks for the issue!

It looks like this is related to GitLab sub projects.

Repos hosted at the top-level like this work for me:

```
pulumi new https://gitlab.com/justinvp/test
```

As do specific templates within the git repo:

```
pulumi new https://gitlab.com/justinvp/test/typescript
```

You could try moving your template to a top-level repository as a workaround.

I'll keep this open to track adding support for GitLag sub groups/projects.

**Comment by yodamad** [comment]
Hi @justinvp 

thank you for the reply
I've tried moving the project to root level, it works fine, so you're right : doesn't work for subgroups ☹️

Not very convenient for a company

**Comment by Alphacharge** [comment]
I was trying to add my own templates in ~/.pulumi/templates but they get deleted again.
Then i found this issue because i thought about doing it with a gitlab repo.
What's the progress on implementing this sub group functionality? @justinvp 
This would be very helpful to provide own templates in the company.

**Comment by tgummerer** [comment]
I think you can do this with the current functionality, though I think we should make it more obvious how to do that.

If you have a repository on GitLab that's named `group/subgroup/repository-name`, you'd have to use `pulumi new https://gitlab.com/group/subgroup/repository-name.git`.  (I just tried this with a testrepository of mine at `pulumi new https://gitlab.com/testgroup880252/subgroup/testproject.git`).

Similarly if the templates sit in a subfolder of this repo, you can use `pulumi new https://gitlab.com/testgroup880252/subgroup/testproject.git/subfolder`.  

On the CLI side it's hard to disambiguate between these option, since e.g. `a/b/c` could mean "use the `a/b/c` repo (which is what is wanted here)", or "use the `a/b` repository at path `c` (which is what we default to).

**Comment by Alphacharge** [comment]
Thank you very much for the quick response.
Following worked fine:
`pulumi new https://<user>:<token>@<gitlab repo url>/subfolder`
for a private gitlab Repo.
So it's already implemented?
Also is there a way to use a own local template? (even if a repo is easier to maintain)
thx

**Comment by Alphacharge** [comment]
Yes, i am. Maybe this could be also a bit more clear. I have never read that anywhere.
Thank's for the help.
